### PR TITLE
Add docs for airgapped upgrades

### PR DESCRIPTION
**Pull request by harsh-px**
Signed-off-by: Harsh Desai <harsh@portworx.com>

### CODE REVIEW

File: content/portworx-install-with-kubernetes/operate-and-maintain-on-kubernetes/upgrade/_index.md [hunk 1]
```diff
@@ -31,30 +26,82 @@ This runs a script that will start a Kubernetes Job to perform the following ope
...
 ```text
-curl -fsL https://install.portworx.com/upgrade | bash -s -- -t 1.7.2
+curl -fsL https://install.portworx.com/2.0/upgrade | bash -s -- -t 2.0.2.2
 ```
 
-#### Disable scaling down of shared Portworx applications during the upgrade {#disable-scaling-down-of-shared-portworx-applications-during-the-upgrade}
+## Airgapped clusters
 
-You can invoke the upgrade script with _–scaledownsharedapps off_ to skip scaling down Deployments and StatefulSets that use shared Portworx PersistentVolumeClaim.
+When upgrading Portworx in Kubernetes using the default curl command, a number of docker images are fetched from registries on the internet. If your nodes don't have access to the standard registries on the internet, you need to first pull the required images in your cluster and then give your image names to the curl command.
```
Comment (zoxpx) [inline content/portworx-install-with-kubernetes/operate-and-maintain-on-kubernetes/upgrade/_index.md]: re "using the default curl command", the "curl command" is awkward, this is normally called "bootstrap" -- please rename to ..."upgrade bootstrap"
re "fetched from registries" -- I'd recommend making this more precise (e.g. I'd use "docker registries" or "container registries")
re "on the internet" -- you're using it as a general noun here, so it should be capitalized ("Internet")
re "standard registries on the internet" -- would be good to list them (e.g. docker.io, gcr.io)

... and finally:
re "give your image names to the curl command" -- sorry, but I have no idea which "curl command" you are referring to (the one from bootstrap?); did you mean "please specify precise image names with the upgrade-bootstrap command" (+ handy example?)

File: content/portworx-install-with-kubernetes/operate-and-maintain-on-kubernetes/upgrade/_index.md [hunk 2]
```diff
@@ -31,30 +26,82 @@ This runs a script that will start a Kubernetes Job to perform the following ope
...
+else
+  echo $REGISTRY | grep -q /
+  if [ $? -eq 0 ]; then
+    TALISMAN_IMAGE=$REGISTRY/talisman
+    OCIMON_IMAGE=$REGISTRY/oci-monitor
+  else
+    TALISMAN_IMAGE=$REGISTRY/portworx/talisman
+    OCIMON_IMAGE=$REGISTRY/portworx/oci-monitor
+  fi
+fi
```
Comment (zoxpx) [inline content/portworx-install-with-kubernetes/operate-and-maintain-on-kubernetes/upgrade/_index.md]: I'd suggest you rearrange the expressions in "define then override" manner.   it'll be much more readable what is used by default  (also over 90% of the time).

```bash
# Default image names
TALISMAN_IMAGE=portworx/talisman
OCIMON_IMAGE=portworx/oci-monitor

# Do we have container registry override?
if [ "x$REGISTRY" != x ]; then
   echo $REGISTRY | grep -q /
   if [ $? -eq 0 ]; then  # REGISTRY defines both registry and repository
      TALISMAN_IMAGE=$REGISTRY/talisman
      OCIMON_IMAGE=$REGISTRY/oci-monitor
   else                   # $REGISTRY contains only registry, we'll assume default repositories
      TALISMAN_IMAGE=$REGISTRY/portworx/talisman
      OCIMON_IMAGE=$REGISTRY/portworx/oci-monitor
   fi
fi
```

File: content/portworx-install-with-kubernetes/operate-and-maintain-on-kubernetes/upgrade/_index.md [hunk 3]
```diff
@@ -31,30 +26,82 @@ This runs a script that will start a Kubernetes Job to perform the following ope
...
-By default, the upgrade process scales down shared applications as that avoids a node reboot when upgrading between major Portworx versions. Disabling that flag would mean the node would need a reboot before Portworx comes up with the new major version.
-{{</info>}}
+Now pull the required Portworx images.
+
+```text
+if [ -z "$NEW_PX_VER" ]; then
+  PX_VER=$(curl -fsL https://install.portworx.com/2.0/upgrade | grep ^OCI_MON_TAG= | awk -F"=" '{print $2}')
+else
+  PX_VER=$NEW_PX_VER
+fi
```
Comment (zoxpx) [inline content/portworx-install-with-kubernetes/operate-and-maintain-on-kubernetes/upgrade/_index.md]: We have to simplify the use of the `NEW_PX_VER`  -- the way it's described here is **very** confusing.

I'd suggest to remove the `NEW_PX_VER` entirely, and just mention one can use/define their own value, i.e.:

```bash
# To determine the latest minor 2.0 release currently available, please use the curl-expression below
# Alternatively, you can specify the version yourself, e.g.: PX_VER=2.0.2.2
PX_VER=$(curl -fs https://install.portworx.com/2.0/upgrade | awk -F'=' '/^OCI_MON_TAG=/{print $2}')
....
```
* note, the `awk` can also do implicit "grepping"

File: content/portworx-install-with-kubernetes/operate-and-maintain-on-kubernetes/upgrade/_index.md [hunk 4]
```diff
@@ -107,3 +154,18 @@ If the upgrade job crashes unexpectedly and fails to restore shared applications
...
+
+For example:
+
+```text
+curl -fsL https://install.portworx.com/2.0/upgrade | bash -s -- --scaledownsharedapps off
+```
+
+{{<info>}}
+**Reboot requirement:**
+By default, the upgrade process scales down shared applications as that avoids a node reboot when upgrading between major Portworx versions. Disabling that flag would mean the node would need a reboot before Portworx comes up with the new major version.
```
Comment (zoxpx) [inline content/portworx-install-with-kubernetes/operate-and-maintain-on-kubernetes/upgrade/_index.md]: Does it do it every time?  If not,  I'd suggest inserting "if/as required" somewhere

File: content/portworx-install-with-kubernetes/on-premise/airgapped/shared/_index.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
+---
+title: airgapped installs shared
+keywords: portworx, ASG, AWS,GCP
```
Comment (zoxpx) [inline content/portworx-install-with-kubernetes/on-premise/airgapped/shared/_index.md]: minor: spacing typo?

File: content/portworx-install-with-kubernetes/operate-and-maintain-on-kubernetes/upgrade/_index.md (hunk 3 above)
Comment (harsh-px) [reply content/portworx-install-with-kubernetes/operate-and-maintain-on-kubernetes/upgrade/_index.md]: done

File: content/portworx-install-with-kubernetes/operate-and-maintain-on-kubernetes/upgrade/_index.md (hunk 1 above)
Comment (harsh-px) [reply content/portworx-install-with-kubernetes/operate-and-maintain-on-kubernetes/upgrade/_index.md]: reworded

File: content/portworx-install-with-kubernetes/operate-and-maintain-on-kubernetes/upgrade/_index.md (hunk 2 above)
Comment (harsh-px) [reply content/portworx-install-with-kubernetes/operate-and-maintain-on-kubernetes/upgrade/_index.md]: done

File: content/portworx-install-with-kubernetes/operate-and-maintain-on-kubernetes/upgrade/_index.md (hunk 4 above)
Comment (harsh-px) [reply content/portworx-install-with-kubernetes/operate-and-maintain-on-kubernetes/upgrade/_index.md]: fixed

**Pull request merged**